default/scripting/techs/defense/mines.macros

```diff
@@ -271,3 +190,17 @@ EG_SYSTEM_MINES
                     empire = Target.Owner
             ]
 '''
+
+SYSTEM_CONTAINS_SOURCE_OR_OWNED_PLANET
+'''                ContainedBy And [
+                    System 
+                    Contains Or [
+                        Source
+                        And [
+                            Planet
+                            OwnedBy empire = Source.Owner
+                            OwnerHasTech name = ThisTech
```

Comment:
OwnerHasTech is included to keep specials from applying empire-wide. This is wasted effort for the techs, attempts at ObjectType did not pan out (bad syntax? `(Source.ObjectType type != Special)`).
`ThisTech` was only chosen due to using it with OwnerHasTech.  Prefer `CurrentContent`?